app/public.js

```diff
@@ -70,13 +72,50 @@
         }
       }
     });
+    setRefreshImagesWatcher();
+  }
+
+  /**
+   * Refresh overlay
+   */
+  function refreshImagesInfo() {
+    hideImagesInfo();
+    showImagesInfo();
+  }
+
+  /**
+   * Set event listeners
+   */
+  function setRefreshImagesWatcher() {
+    let width = window.innerWidth;
+    let height = window.innerHeight;
+    let path = window.location.pathname;
+    let update;
+
+    refreshImagesInterval = setInterval(() => {
+      update = false;
+      /* istanbul ignore if */
+      if (window.location.pathname !== path) {
+        update = true;
+        path = window.location.pathname;
```

Comment:
@tabanliviu DoD - This doesn't include changes in the hash (either for traditional hash or non-HTML5 SPAs).